### PR TITLE
Check if another resource exists when creating an alias

### PR DESCRIPTION
**Pull request by jsoriano**
When creating an alias, if an index exists with the same name, alias is
not created assuming that it already exists. This can create some
confusion in some cases, as users can see that setup works, but indexes
and aliases are not correctly configured afterwards, leading to field
mapping issues and other kinds of problems.

This change makes beats explictly check if the resource that already
exists is actually an alias, if not, it returns a different failure that
can be seen by the user when running setup.

### CODE REVIEW

**Comment by jsoriano** [comment]
@urso I have added a check also to `HasAlias`. I am keeping the check also in `CreateAlias` because it shouldn't succeed if it is not able to create the alias. Let me know what you think.

Regarding using the `cat` API, I think it is better not to use it. For example it responses with status 200 (and an empty response) when a requested resource doesn't exist, this kind of things would force us to parse the responses.

**Comment by jsoriano** [comment]
jenkins, test this again please

File: libbeat/idxmgmt/ilm/client_handler.go [hunk 1]
```diff
@@ -179,6 +193,14 @@ func (h *ESClientHandler) CreateAlias(alias Alias) error {
...
 	status, res, err := h.client.Request("PUT", "/"+firstIndex, "", nil, body)
 	if status == 400 {
+		hasAlias, err := h.HasAlias(alias.Name)
+		if err != nil {
+			return err
+		}
+		if !hasAlias {
+			return errf(ErrInvalidAlias,
+				"resource '%v' exists, but it is not an alias", alias.Name)
+		}
```
Comment (urso) [inline libbeat/idxmgmt/ilm/client_handler.go]: the later error 'ErrInvalidAlias' with message is now returned by 'HasAlias' ;)

Comment (jsoriano) [reply libbeat/idxmgmt/ilm/client_handler.go]: Ok, removed the check here :)

**Comment by jsoriano** [comment]
jenkins, test this again please

**Comment by jsoriano** [comment]
Failing tests are not related, merging this.

**Pull request merged**